test/unit/controller.js

```diff
@@ -838,3 +838,114 @@ exports['controller.closeTesselConnections'] = {
       });
   },
 };
+
+exports['controller.root'] = {
+  setUp: function(done) {
+    this.sandbox = sinon.sandbox.create();
+    this.logsWarn = this.sandbox.stub(logs, 'warn', function() {});
+    this.logsInfo = this.sandbox.stub(logs, 'info', function() {});
+    this.logsBasic = this.sandbox.stub(logs, 'basic', function() {});
+
+    this.standardTesselCommand = this.sandbox.stub(controller, 'standardTesselCommand').returns(Promise.resolve());
+    done();
+  },
+
+  tearDown: function(done) {
+    this.sandbox.restore();
+    done();
+  },
+
+  setsOptsLanTrue: function(test) {
+    test.expect(3);
+
+    var opts = {};
+
+    controller.root(opts)
```

Comment:
Oh I see - because `standardTesselCommand` returns Promise.resolve (instead of an unfulfilled promise), the whole `root` procedure finishes up before it gets to the `shell` part.